include/mbedtls/config.h

```diff
@@ -29,9 +29,20 @@
 #ifndef MBEDTLS_CONFIG_H
 #define MBEDTLS_CONFIG_H
 
-#if defined(_MSC_VER) && !defined(_CRT_SECURE_NO_DEPRECATE)
+#ifdef _MSC_VER
+#ifndef _CRT_SECURE_NO_DEPRECATE
 #define _CRT_SECURE_NO_DEPRECATE 1
 #endif
+#include <winsock2.h>
+#include <tchar.h>
+#define MBEDTLS__T _T
+#define MBEDTLS_TCHAR TCHAR
+#define MBEDTLS_FOPEN _tfopen
+#else
+#define MBEDTLS__T(x) x
```

Comment:
What is `MBEDTLS__T` supposed to mean? and `MBEDTLS_TCHAR`? Please use names that are meaningful on their own. Most Mbed TLS developers aren't familiar with the Windows APIs. Also, please document when to use those macros.

Reply:
That is a tough job to comment on Windows-specific changes without knowing the platform.

> What is MBEDTLS__T supposed to mean? and MBEDTLS_TCHAR?

Here is a short article covering the basics: https://msdn.microsoft.com/en-us/library/dybsewaf.aspx

> Please use names that are meaningful on their own.

That might be redirected to Microsoft.
In the fist commit I tried to use original macro names defined in Windows, but had to comply to local standards and to add prefixes.

> when to use those macros.

In short, do not use it in non-Windows code, because after macro expansion this code should be the same  as it was before the requested changes were applied. That is why `TCHAR` becomes simply char and `_T()` is a no-op in non-Windows environment.

`Note. This could be a duplicate, but probably a wrong button was pressed and the comment was lost.`